### PR TITLE
Take a copy of existing TACACS credentials and restore it during upgrade

### PR DESCRIPTION
**Pull request by renukamanavalan**
In scenario where upgrade gets config from minigraph, it could miss tacacs credentials as they are not in minigraph. Hence restore explicitly upon load-minigraph, if  present.

**- Why I did it**
Upon boot, when config migration is required, the switch could load config from minigraph. The config-load from minigraph would wipe off TACACS key and disable login via TACACS, which would disable all remote user access. This change, would re-configure the TACACS if there is a saved copy available.

**- How I did it**
When config is loaded from minigraph, look for a TACACS credentials back up (tacacs.json) under /etc/sonic/old_config. If present, load the credentials into running config, before config-save is called.

**- How to verify it**
Remove /etc/sonic/config_db.json and do an image update. Upon reboot, w/o this change, you would not be able ssh in as remote user. You may login as admin and check out, "show tacacs" & "show aaa" to verify that tacacs-key is missing and login is not enabled for tacacs.
With this change applied, remove /etc/sonic/config_db.json, but save tacacs & aaa credentials as tacacs.json in /etc/sonic/. Upon reboot, you should see remote user access possible.

**- Which release branch to backport (provide reason below if selected)**

- [x] 201811
- [x] 201911
- [x] 202006
- [x] 202012

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: files/image_config/config-setup/config-setup [hunk 1]
```diff
@@ -355,6 +362,10 @@ do_config_backup()
     echo "Taking backup of curent configuration"
     rm -rf /host/old_config
     cp -ar /etc/sonic /host/old_config
+
+    # Save TACACS
+    sonic-cfggen -d --preset tac > /host/old_config/${TACACS_JSON_BACKUP}
```
Comment (lguohan) [inline files/image_config/config-setup/config-setup:367]: i am not sure if we need this as the config_db has the tacacs cofiguration.

Comment (renukamanavalan) [reply files/image_config/config-setup/config-setup]: dropped

**Comment by lguohan** [comment]
@renukamanavalan , can you fill the questionaire?

**Comment by renukamanavalan** [comment]
> @renukamanavalan , can you fill the questionaire?

Done

File: files/image_config/config-setup/config-setup [hunk 2]
```diff
@@ -108,6 +110,11 @@ reload_minigraph()
 {
     echo "Reloading minigraph..."
     config load_minigraph -y -n 
+    if [ -f /etc/sonic/old_config/${TACACS_JSON_BACKUP} ]; then
```
Comment (yxieca) [inline files/image_config/config-setup/config-setup:113]: Minor improvement: you could test -r instead of -f.

**Review by yxieca — APPROVED** [review]

**Comment by renukamanavalan** [comment]
retest vs please

**Comment by renukamanavalan** [comment]
retest vsimage please

**Comment by renukamanavalan** [comment]
retest vsimage please

**Comment by yxieca** [comment]
@renukamanavalan can you open an PR for 201811 branch? This change cannot be cherry-picked cleanly.

**Pull request merged**